checks/evaluation/pinned_dependencies.go

```diff
@@ -95,132 +75,28 @@ func probeRemToRuleRem(rem *probe.Remediation) *rule.Remediation {
-				LineStart: &rr.Location.Offset,
-				LineEnd:   &rr.Location.EndOffset,
-				Snippet:   &rr.Location.Snippet,
-			}
-			f := &finding.Finding{
-				Probe:    "",
-				Outcome:  finding.OutcomeNotApplicable,
-				Message:  *rr.Msg,
-				Location: loc,
-			}
-			findings = append(findings, *f)
-			continue
-		}
-		if rr.Pinned == nil {
-			loc := &finding.Location{
-				Type:      rr.Location.Type,
-				Path:      rr.Location.Path,
-				LineStart: &rr.Location.Offset,
-				LineEnd:   &rr.Location.EndOffset,
-				Snippet:   &rr.Location.Snippet,
-			}
-			f := &finding.Finding{
-				Probe:    "",
-				Outcome:  finding.OutcomeNotApplicable,
-				Message:  fmt.Sprintf("%s has empty Pinned field", rr.Type),
-				Location: loc,
-			}
-			findings = append(findings, *f)
-			continue
-		}
-		if !*rr.Pinned {
-			loc := &finding.Location{
-				Type:      rr.Location.Type,
-				Path:      rr.Location.Path,
-				LineStart: &rr.Location.Offset,
-				LineEnd:   &rr.Location.EndOffset,
-				Snippet:   &rr.Location.Snippet,
-			}
-			f := &finding.Finding{
-				Probe:    "",
-				Outcome:  finding.OutcomeNegative,
-				Message:  generateTextUnpinned(&rr),
-				Location: loc,
-			}
-			if rr.Remediation != nil {
-				f.Remediation = ruleRemToProbeRem(rr.Remediation)
-			}
-			f = f.WithValues(map[string]int{
-				depTypeKey: dependencyTypes[rr.Type],
-			})
-			findings = append(findings, *f)
-		} else {
-			loc := &finding.Location{
-				Type:      rr.Location.Type,
-				Path:      rr.Location.Path,
-				LineStart: &rr.Location.Offset,
-				LineEnd:   &rr.Location.EndOffset,
-				Snippet:   &rr.Location.Snippet,
-			}
-			f := &finding.Finding{
-				Probe:    "",
-				Outcome:  finding.OutcomePositive,
-				Location: loc,
-			}
-			f = f.WithValues(map[string]int{
-				depTypeKey: dependencyTypes[rr.Type],
-			})
-			findings = append(findings, *f)
-		}
-	}
-	return findings, nil
-}
-
 // PinningDependencies applies the score policy for the Pinned-Dependencies check.
-func PinningDependencies(name string, c *checker.CheckRequest,
-	r *checker.PinningDependenciesData,
+func PinningDependencies(name string,
+	findings []finding.Finding,
+	dl checker.DetailLogger,
 ) checker.CheckResult {
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+	expectedProbes := []string{
+		pinsDependencies.Probe,
+	}
+
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
 		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
 	var wp workflowPinningResult
 	pr := make(map[checker.DependencyUseType]pinnedResult)
-	dl := c.Dlogger
-
-	findings, err := dependenciesToFindings(r)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
 
 	for i := range findings {
 		f := findings[i]
 		switch f.Outcome {
+		case finding.OutcomeNotAvailable:
+			return checker.CreateInconclusiveResult(name, "no dependencies found")
```

Comment:
optional: I think there are few mismatched outcomes -> behavior. But as a purely conversion PR this isn't necessarily in scope on this PR.

if you'd like to tackle it here, we should briefly discuss changes before switching them.

Reply:
I think the `finding.OutcomeNotAvailable` and `finding.OutcomeNotApplicable` would be swapped?

Reply:
Let's defer this to a later PR. I have created a tracker here: https://github.com/ossf/scorecard/issues/3855